# Data Preparation and Cleaning

Import essential libraries

In [469]:
import pandas as pd
from IPython.display import display

---

## Dataset 1 - League of Legends - 2024 Competitive Game Dataset

Checking the match's data and how many columns are in the dataset.

In [470]:
raw_match_data = pd.read_csv('datasets/2024_LoL_esports_match_data_from_OraclesElixir_gamedata.csv', sep=',')
display(raw_match_data)

gameid datacompleteness league  year       split  playoffs  \
0     10660-10660_game_3          partial   DCup  2023         NaN         0   
1        LOLTMNT05_13659         complete  LPLOL  2024      Spring         0   
2        LOLTMNT04_24264         complete    LEC  2024      Winter         1   
3        LOLTMNT01_52094         complete  ESLOL  2024      Spring         0   
4        LOLTMNT02_29074         complete    LCK  2024      Spring         0   
...                  ...              ...    ...   ...         ...       ...   
1018     LOLTMNT05_13780         complete    LEC  2024      Winter         0   
1019  10698-10698_game_1          partial    LPL  2024      Spring         0   
1020     LOLTMNT02_33054         complete    LJL  2024      Spring         0   
1021  10752-10752_game_1          partial    LDL  2024  Placements         0   
1022     LOLTMNT02_23326         complete    PCS  2024      Spring         0   

                 date  game  patch  gamelength    ckpm  \
0       1/1/2024 6:54     3  13.24        1324  1.0876   
1     26/1/2024 19:10     1  14.01        1364  0.7918   
2      3/2/2024 18:47     2  14.02        1350  1.0222   
3      8/2/2024 19:11     1  14.02        1357  1.4591   
4      24/1/2024 9:00     2  14.01        1283  1.0756   
...               ...   ...    ...         ...     ...   
1018  28/1/2024 16:13     1  14.01        2963  0.9720   
1019   4/2/2024 11:19     1  14.01        3180  0.3774   
1020  27/1/2024 11:24     2  14.01        2950  0.9559   
1021    3/2/2024 8:19     1  14.01        2923  1.1906   
1022  19/1/2024 13:28     2  14.01        3226  0.9485   

                     t1_name                                    t1_id  \
0                LNG Esports  oe:team:a9145b7711873f53e610fbba0493484   
1                EGN Esports  oe:team:8d48390319b4f7c25c13dd410b3a0bf   
2                 G2 Esports  oe:team:7d6673d3a9d00363c6bebc1a630da6e   
3            One Way Esports  oe:team:a181ac52921fccf36966206cb6591b0   
4        Hanwha Life Esports  oe:team:3a1d18f46bcb3716ebcfcf4ef068934   
...                      ...                                      ...   
1018           Team Heretics  oe:team:1285e1c04d54de96502f12da4d49458   
1019     Royal Never Give Up  oe:team:73f3126d91d35faa5d5ce34abadb865   
1020              V3 Esports  oe:team:af4b23d4634c4e6ecb5659fb1e689e9   
1021  Top Esports Challenger  oe:team:eaeaad6cbcd022d2b872d4776aae2e6   
1022       Deep Cross Gaming  oe:team:7e56e9367c023df96cc8257ff4e16c0   

        t1_ban1     t1_ban2   t1_ban3    t1_ban4    t1_ban5  t1_result  \
0          Rell    Nocturne  Tristana  Jarvan IV     Rumble          1   
1     Seraphine       Karma      Ashe   Nautilus       Rell          1   
2        Rumble  Blitzcrank      Gnar    Orianna    LeBlanc          1   
3         Neeko        Azir      Udyr    Orianna      Fiora          1   
4         Varus       Senna        Vi      Brand       Yone          1   
...         ...         ...       ...        ...        ...        ...   
1018    K'Sante      Maokai   LeBlanc        Jax     Rumble          0   
1019     Draven       Milio    Rumble   Nocturne  Jarvan IV          0   
1020       Ashe        Udyr  Nocturne        Jax    Sejuani          0   
1021        NaN         NaN       NaN        NaN        NaN          0   
1022    Lee Sin     LeBlanc  Nocturne    Sejuani   Xin Zhao          1   

      t1_kills  t1_deaths  t1_kpm t1_side  t1_doublekills  t1_triplekills  \
0           21          3  0.9517    Blue             NaN             NaN   
1           15          3  0.6598    Blue             3.0             1.0   
2           21          2  0.9333    Blue             2.0             0.0   
3           26          8  1.1496    Blue             5.0             0.0   
4           20          3  0.9353    Blue             5.0             0.0   
...        ...        ...     ...     ...             ...             ...   
1018        25         23  0.5062    Blue             3

Printing all columns to check which columns suits our problem definination.

In [471]:
print(raw_match_data.columns.tolist())

['gameid', 'datacompleteness', 'league', 'year', 'split', 'playoffs', 'date', 'game', 'patch', 'gamelength', 'ckpm', 't1_name', 't1_id', 't1_ban1', 't1_ban2', 't1_ban3', 't1_ban4', 't1_ban5', 't1_result', 't1_kills', 't1_deaths', 't1_kpm', 't1_side', 't1_doublekills', 't1_triplekills', 't1_quadrakills', 't1_pentakills', 't1_firstblood', 't1_firstdragon', 't1_dragons', 't1_elementaldrakes', 't1_infernals', 't1_mountains', 't1_clouds', 't1_oceans', 't1_chemtechs', 't1_hextechs', 't1_elders', 't1_firstherald', 't1_heralds', 't1_firstbaron', 't1_barons', 't1_firsttower', 't1_towers', 't1_firstmidtower', 't1_firsttothreetowers', 't1_turretplates', 't1_inhibitors', 't1_damagetochampions', 't1_dpm', 't1_damageshare', 't1_damagetakenperminute', 't1_damagemitigatedperminute', 't1_wardsplaced', 't1_wpm', 't1_wardskilled', 't1_wcpm', 't1_controlwardsbought', 't1_visionscore', 't1_vspm', 't1_totalgold', 't1_earnedgold', 't1_earned_gpm', 't1_earnedgoldshare', 't1_goldspent', 't1_gspd', 't1_totalcs'

After checking through the columns, we conclude these columns will be useful for our **main problem definition** as these influence the match result, where the columns are the result of the match between two teams, as well as the champions they used for each player.

```
t1_result
t2_result
t1p1_champion
t1p2_champion
t1p3_champion
t1p4_champion
t1p5_champion
t2p1_champion
t2p2_champion
t2p3_champion
t2p4_champion
t2p5_champion
```

And these columns for our **2nd problem definition** as these influence the gold earned, where the columns show the total amount of gold earned for each team, number of minion killed, as well as the objectives they completed.

```
t1_totalgold
t2_totalgold
t1_barons
t2_barons
t1_dragons
t2_dragons
t1_heralds
t2_heralds
t1_towers
t2_towers
t1_inhibitors
t2_inhibitors
t1_kills
t2_kills
t1_totalcs
t2_totalcs
```

In [472]:
desired_columns_dataset_1_problem_1 = ['gameid', 't1_result', 't2_result', 't1p1_champion', 't1p2_champion', 't1p3_champion', 't1p4_champion', 't1p5_champion', 't2p1_champion', 't2p2_champion', 't2p3_champion', 't2p4_champion', 't2p5_champion']

desired_columns_dataset_1_problem_2 = ['t1_totalgold', 't2_totalgold', 't1_barons', 't2_barons', 't1_dragons', 't2_dragons', 't1_heralds', 't2_heralds', 't1_towers', 't2_towers', 't1_inhibitors', 't2_inhibitors', 't1_kills', 't2_kills', 't1_totalcs', 't2_totalcs']

clean_match_data = pd.DataFrame(raw_match_data, columns=(desired_columns_dataset_1_problem_1 + desired_columns_dataset_1_problem_2))
display(clean_match_data)

gameid  t1_result  t2_result t1p1_champion t1p2_champion  \
0     10660-10660_game_3          1          0           Jax      Bel'Veth   
1        LOLTMNT05_13659          1          0          Udyr      Xin Zhao   
2        LOLTMNT04_24264          1          0           Zac            Vi   
3        LOLTMNT01_52094          1          0        Aatrox           Jax   
4        LOLTMNT02_29074          1          0        Rumble        Maokai   
...                  ...        ...        ...           ...           ...   
1018     LOLTMNT05_13780          0          1          Udyr       Lee Sin   
1019  10698-10698_game_1          0          1           Jax       Sejuani   
1020     LOLTMNT02_33054          0          1        Rumble      Xin Zhao   
1021  10752-10752_game_1          0          1        Aatrox      Xin Zhao   
1022     LOLTMNT02_23326          1          0   Mordekaiser            Vi   

     t1p3_champion t1p4_champion t1p5_champion t2p1_champion t2p2_champion  \
0            Neeko       Caitlyn           Lux        Aatrox       Lee Sin   
1             Azir       Kalista  Renata Glasc       K'Sante      Nocturne   
2             Azir         Senna        Maokai        Aatrox      Xin Zhao   
3             Hwei       Kalista      Nautilus        Darius      Xin Zhao   
4         Tristana        Lucian         Milio          Udyr       Lee Sin   
...            ...           ...           ...           ...           ...   
1018         Akali       Kalista          Hwei          Gnar          Rell   
1019          Azir         Varus  Renata Glasc       K'Sante        Maokai   
1020         Akali  Miss Fortune         Neeko       K'Sante       Lee Sin   
1021         Corki         Xayah         Rakan       K'Sante           Jax   
1022          Azir        Lucian         Milio        Irelia         Viego   

     t2p3_champion t2p4_champion t2p5_champion  t1_totalgold  t2_totalgold  \
0             Azir       Kalista  Renata Glasc         51091         30710   
1              Lux         Ziggs         Janna         49451         32225   
2            Sylas        Ezreal         Yuumi         50055         32273   
3            Akali        Lucian         Milio         51601         32965   
4             Azir       Kalista  Renata Glasc         47569         33183   
...            ...           ...           ...           ...           ...   
1018        Kai'Sa         Varus  Renata Glasc         92016         93497   
1019       LeBlanc       Kalista          Zyra         92082         93518   
1020          Azir         Senna      Nautilus         87286         93730   
1021       Orianna       Kalista          Rell         92168         95579   
1022       Orianna       Caitlyn           Lux        100012        100353   

      t1_barons  t2_barons  t1_dragons  t2_dragons  t1_heralds  t2_heralds  \
0             1          0           2           1           0           0   
1             1          0           3           0           1           0   
2             0          0           3           0           1           0   
3             1          0           1           1           1           0   
4             0          0           2           1           1           0   
...         ...        ...         ...         ...         ...         ...   
1018          3          0           3           4           1           0   
1019          2          0           3           4           0           0   
1020          1          1           4           3           1           0   
1021          3          0           3           4           0           0   
1022          0          4           3           5           0           1   

      t1_towers  t2_towers  t1_inhibitors  t2_inhibitors  t1_kills  t2_kills  \
0            10          0              2              0        21         3   
1             9          9              1              1        15         3   
2            10         10    

The clean version of the 1st dataset, which will be easier to work with when compared to the raw version.

---

## Dataset 2 - Champions Dataset

Checking the data points and how many columns are in the dataset.

In [473]:
raw_champion_data = pd.read_csv('datasets/LoL-Champions.csv', sep=',')
display(raw_champion_data)

Id     Name             Class  Style  Difficulty DamageType  Damage  \
0      1   Aatrox           Warrior      3           2          P       3   
1      2     Ahri              Mage     10           2          M       3   
2      3    Akali          Assassin      7           2         PM       3   
3      4  Alistar      Tank-Support      7           1          M       1   
4      5    Amumu              Tank      9           1          M       2   
..   ...      ...               ...    ...         ...        ...     ...   
162  163    Milio           Support      8           1          M       1   
163  164  Naafiri          Assassin      4           1          P       3   
164  165    Briar  Assassin-Warrior      3           1          P       3   
165  166     Hwei              Mage     10           3          M       3   
166  167  Smolder          Marksman      3           2          P       3   

     Sturdiness  Crowd-Control  Mobility  Functionality  
0             3              2         2            2.0  
1             1              2         3            1.0  
2             1              1         3            1.0  
3             3              3         1            2.0  
4             3              3         1            1.0  
..          ...            ...       ...            ...  
162           1              3         1            3.0  
163           2              1         2            1.0  
164           2              2         3            1.0  
165           1              3         1            3.0  
166           1              1         2            2.0  

[167 rows x 11 columns]

In [474]:
print(raw_champion_data.columns.tolist())

['Id', 'Name', 'Class', 'Style', 'Difficulty', 'DamageType', 'Damage', 'Sturdiness', 'Crowd-Control', 'Mobility', 'Functionality']


Not much columns when compared to the first data sets, but we only needed the following columns to tackle our main problem defination, since we are trying to find champion composition (based on class) that maximize the chances of winning.

```
Name
Class
```

Also since there is 167 rows, it also means that there is **167 champions** in the League of Legends game currently.

In [475]:
desired_columns_dataset_2 = ['Name', 'Class']

clean_champion_data = pd.DataFrame(raw_champion_data, columns=desired_columns_dataset_2)
display(clean_champion_data)

Name             Class
0     Aatrox           Warrior
1       Ahri              Mage
2      Akali          Assassin
3    Alistar      Tank-Support
4      Amumu              Tank
..       ...               ...
162    Milio           Support
163  Naafiri          Assassin
164    Briar  Assassin-Warrior
165     Hwei              Mage
166  Smolder          Marksman

[167 rows x 2 columns]

The clean version of the 2nd dataset.

---

## Merging Both Datasets

We will merge the clean_match_data with clean_champions_data based on the name of the champion, and then adding the class of that champion for each champion.

In [476]:
# dataset 1 - clean version
display(clean_match_data)

gameid  t1_result  t2_result t1p1_champion t1p2_champion  \
0     10660-10660_game_3          1          0           Jax      Bel'Veth   
1        LOLTMNT05_13659          1          0          Udyr      Xin Zhao   
2        LOLTMNT04_24264          1          0           Zac            Vi   
3        LOLTMNT01_52094          1          0        Aatrox           Jax   
4        LOLTMNT02_29074          1          0        Rumble        Maokai   
...                  ...        ...        ...           ...           ...   
1018     LOLTMNT05_13780          0          1          Udyr       Lee Sin   
1019  10698-10698_game_1          0          1           Jax       Sejuani   
1020     LOLTMNT02_33054          0          1        Rumble      Xin Zhao   
1021  10752-10752_game_1          0          1        Aatrox      Xin Zhao   
1022     LOLTMNT02_23326          1          0   Mordekaiser            Vi   

     t1p3_champion t1p4_champion t1p5_champion t2p1_champion t2p2_champion  \
0            Neeko       Caitlyn           Lux        Aatrox       Lee Sin   
1             Azir       Kalista  Renata Glasc       K'Sante      Nocturne   
2             Azir         Senna        Maokai        Aatrox      Xin Zhao   
3             Hwei       Kalista      Nautilus        Darius      Xin Zhao   
4         Tristana        Lucian         Milio          Udyr       Lee Sin   
...            ...           ...           ...           ...           ...   
1018         Akali       Kalista          Hwei          Gnar          Rell   
1019          Azir         Varus  Renata Glasc       K'Sante        Maokai   
1020         Akali  Miss Fortune         Neeko       K'Sante       Lee Sin   
1021         Corki         Xayah         Rakan       K'Sante           Jax   
1022          Azir        Lucian         Milio        Irelia         Viego   

     t2p3_champion t2p4_champion t2p5_champion  t1_totalgold  t2_totalgold  \
0             Azir       Kalista  Renata Glasc         51091         30710   
1              Lux         Ziggs         Janna         49451         32225   
2            Sylas        Ezreal         Yuumi         50055         32273   
3            Akali        Lucian         Milio         51601         32965   
4             Azir       Kalista  Renata Glasc         47569         33183   
...            ...           ...           ...           ...           ...   
1018        Kai'Sa         Varus  Renata Glasc         92016         93497   
1019       LeBlanc       Kalista          Zyra         92082         93518   
1020          Azir         Senna      Nautilus         87286         93730   
1021       Orianna       Kalista          Rell         92168         95579   
1022       Orianna       Caitlyn           Lux        100012        100353   

      t1_barons  t2_barons  t1_dragons  t2_dragons  t1_heralds  t2_heralds  \
0             1          0           2           1           0           0   
1             1          0           3           0           1           0   
2             0          0           3           0           1           0   
3             1          0           1           1           1           0   
4             0          0           2           1           1           0   
...         ...        ...         ...         ...         ...         ...   
1018          3          0           3           4           1           0   
1019          2          0           3           4           0           0   
1020          1          1           4           3           1           0   
1021          3          0           3           4           0           0   
1022          0          4           3           5           0           1   

      t1_towers  t2_towers  t1_inhibitors  t2_inhibitors  t1_kills  t2_kills  \
0            10          0              2              0        21         3   
1             9          9              1              1        15         3   
2            10         10    

In [477]:
# dataset 2 - clean version
display(clean_champion_data)

Name             Class
0     Aatrox           Warrior
1       Ahri              Mage
2      Akali          Assassin
3    Alistar      Tank-Support
4      Amumu              Tank
..       ...               ...
162    Milio           Support
163  Naafiri          Assassin
164    Briar  Assassin-Warrior
165     Hwei              Mage
166  Smolder          Marksman

[167 rows x 2 columns]

In [478]:
# merging both datasets using for loop
for i in range(1, 3):
    team = "t" + str(i)

    for j in range(1, 6):
        position = "p" + str(j)
        champion = team + position + "_champion"
        champion_class = team + position + "_class"

        if i == 1 and j == 1:
            merged_data = clean_match_data

        merged_data = pd.merge(merged_data, clean_champion_data, left_on=champion, right_on='Name', how='left')
        merged_data.rename(columns={'Class': champion_class}, inplace=True)
        merged_data.drop('Name', axis=1, inplace=True)

display(merged_data)

gameid  t1_result  t2_result t1p1_champion t1p2_champion  \
0     10660-10660_game_3          1          0           Jax      Bel'Veth   
1        LOLTMNT05_13659          1          0          Udyr      Xin Zhao   
2        LOLTMNT04_24264          1          0           Zac            Vi   
3        LOLTMNT01_52094          1          0        Aatrox           Jax   
4        LOLTMNT02_29074          1          0        Rumble        Maokai   
...                  ...        ...        ...           ...           ...   
1018     LOLTMNT05_13780          0          1          Udyr       Lee Sin   
1019  10698-10698_game_1          0          1           Jax       Sejuani   
1020     LOLTMNT02_33054          0          1        Rumble      Xin Zhao   
1021  10752-10752_game_1          0          1        Aatrox      Xin Zhao   
1022     LOLTMNT02_23326          1          0   Mordekaiser            Vi   

     t1p3_champion t1p4_champion t1p5_champion t2p1_champion t2p2_champion  \
0            Neeko       Caitlyn           Lux        Aatrox       Lee Sin   
1             Azir       Kalista  Renata Glasc       K'Sante      Nocturne   
2             Azir         Senna        Maokai        Aatrox      Xin Zhao   
3             Hwei       Kalista      Nautilus        Darius      Xin Zhao   
4         Tristana        Lucian         Milio          Udyr       Lee Sin   
...            ...           ...           ...           ...           ...   
1018         Akali       Kalista          Hwei          Gnar          Rell   
1019          Azir         Varus  Renata Glasc       K'Sante        Maokai   
1020         Akali  Miss Fortune         Neeko       K'Sante       Lee Sin   
1021         Corki         Xayah         Rakan       K'Sante           Jax   
1022          Azir        Lucian         Milio        Irelia         Viego   

     t2p3_champion t2p4_champion t2p5_champion  t1_totalgold  t2_totalgold  \
0             Azir       Kalista  Renata Glasc         51091         30710   
1              Lux         Ziggs         Janna         49451         32225   
2            Sylas        Ezreal         Yuumi         50055         32273   
3            Akali        Lucian         Milio         51601         32965   
4             Azir       Kalista  Renata Glasc         47569         33183   
...            ...           ...           ...           ...           ...   
1018        Kai'Sa         Varus  Renata Glasc         92016         93497   
1019       LeBlanc       Kalista          Zyra         92082         93518   
1020          Azir         Senna      Nautilus         87286         93730   
1021       Orianna       Kalista          Rell         92168         95579   
1022       Orianna       Caitlyn           Lux        100012        100353   

      t1_barons  t2_barons  t1_dragons  t2_dragons  t1_heralds  t2_heralds  \
0             1          0           2           1           0           0   
1             1          0           3           0           1           0   
2             0          0           3           0           1           0   
3             1          0           1           1           1           0   
4             0          0           2           1           1           0   
...         ...        ...         ...         ...         ...         ...   
1018          3          0           3           4           1           0   
1019          2          0           3           4           0           0   
1020          1          1           4           3           1           0   
1021          3          0           3           4           0           0   
1022          0          4           3           5           0           1   

      t1_towers  t2_towers  t1_inhibitors  t2_inhibitors  t1_kills  t2_kills  \
0            10          0              2              0        21         3   
1             9          9              1              1        15         3   
2            10         10    

The merged dataset is correct, but we will reorder the columns so that the respective columns will be after the respective champion for better clarity.

In [479]:
columns_order = ['t1_result', 't2_result', 't1p1_champion', 't1p1_class', 't1p2_champion', 't1p2_class', 't1p3_champion', 't1p3_class', 't1p4_champion', 't1p4_class', 't1p5_champion', 't1p5_class', 't2p1_champion', 't2p1_class', 't2p2_champion', 't2p2_class', 't2p3_champion', 't2p3_class', 't2p4_champion', 't2p4_class', 't2p5_champion', 't2p5_class', 't1_totalgold', 't2_totalgold', 't1_barons', 't2_barons', 't1_dragons', 't2_dragons', 't1_heralds', 't2_heralds', 't1_towers', 't2_towers', 't1_inhibitors', 't2_inhibitors', 't1_kills', 't2_kills', 't1_totalcs', 't2_totalcs']

merged_data = merged_data[columns_order]

display(merged_data)

t1_result  t2_result t1p1_champion    t1p1_class t1p2_champion  \
0             1          0           Jax       Warrior      Bel'Veth   
1             1          0          Udyr       Warrior      Xin Zhao   
2             1          0           Zac  Tank-Warrior            Vi   
3             1          0        Aatrox       Warrior           Jax   
4             1          0        Rumble  Mage-Warrior        Maokai   
...         ...        ...           ...           ...           ...   
1018          0          1          Udyr       Warrior       Lee Sin   
1019          0          1           Jax       Warrior       Sejuani   
1020          0          1        Rumble  Mage-Warrior      Xin Zhao   
1021          0          1        Aatrox       Warrior      Xin Zhao   
1022          1          0   Mordekaiser       Warrior            Vi   

            t1p2_class t1p3_champion t1p3_class t1p4_champion  \
0              Warrior         Neeko       Mage       Caitlyn   
1              Warrior          Azir       Mage       Kalista   
2              Warrior          Azir       Mage         Senna   
3              Warrior          Hwei       Mage       Kalista   
4                 Tank      Tristana   Marksman        Lucian   
...                ...           ...        ...           ...   
1018  Warrior-Assassin         Akali   Assassin       Kalista   
1019              Tank          Azir       Mage         Varus   
1020           Warrior         Akali   Assassin  Miss Fortune   
1021           Warrior         Corki   Marksman         Xayah   
1022           Warrior          Azir       Mage        Lucian   

            t1p4_class t1p5_champion    t1p5_class t2p1_champion  \
0             Marksman           Lux          Mage        Aatrox   
1             Marksman  Renata Glasc       Support       K'Sante   
2     Marksman-Support        Maokai          Tank        Aatrox   
3             Marksman      Nautilus  Tank-Support        Darius   
4             Marksman         Milio       Support          Udyr   
...                ...           ...           ...           ...   
1018          Marksman          Hwei          Mage          Gnar   
1019          Marksman  Renata Glasc       Support       K'Sante   
1020          Marksman         Neeko          Mage       K'Sante   
1021          Marksman         Rakan       Support       K'Sante   
1022          Marksman         Milio       Support        Irelia   

        t2p1_class t2p2_champion        t2p2_class t2p3_champion  \
0          Warrior       Lee Sin  Warrior-Assassin          Azir   
1     Tank-Warrior      Nocturne          Assassin           Lux   
2          Warrior      Xin Zhao           Warrior         Sylas   
3     Tank-Warrior      Xin Zhao           Warrior         Akali   
4          Warrior       Lee Sin  Warrior-Assassin          Azir   
...            ...           ...               ...           ...   
1018       Warrior          Rell      Tank-Support        Kai'Sa   
1019  Tank-Warrior        Maokai              Tank       LeBlanc   
1020  Tank-Warrior       Lee Sin  Warrior-Assassin          Azir   
1021  Tank-Warrior           Jax           Warrior       Orianna   
1022       Warrior         Viego           Warrior       Orianna   

         t2p3_class t2p4_champion        t2p4_class t2p5_champion  \
0              Mage       Kalista          Marksman  Renata Glasc   
1              Mage         Ziggs              Mage         Janna   
2      Mage-Warrior        Ezreal     Mage-Marksman         Yuumi   
3          Assassin        Lucian          Marksman         Milio   
4              Mage       Kalista          Marksman  Renata Glasc   
...             ...           ...               ...           ...   
1018       Marksman         Varus          Marksman  Renata Glasc   
1019  Mage-Assassin       Kalista          Marksman          Zyra   
1020           Mage         Senna  Marksman-Support      Nautilus   
1021           Mage       Kalista   

The merged dataset is in the correct order and we will create a new csv file based on that dataset (and for further processing if needed) in the EDA section.

In [480]:
merged_data.to_csv("datasets/cleaned_dataset.csv", index=False)

Just doing a final check on the cleaned dataset file.

In [481]:
cleaned_dataset = pd.read_csv('datasets/cleaned_dataset.csv', sep=',')
display(cleaned_dataset)

t1_result  t2_result t1p1_champion    t1p1_class t1p2_champion  \
0             1          0           Jax       Warrior      Bel'Veth   
1             1          0          Udyr       Warrior      Xin Zhao   
2             1          0           Zac  Tank-Warrior            Vi   
3             1          0        Aatrox       Warrior           Jax   
4             1          0        Rumble  Mage-Warrior        Maokai   
...         ...        ...           ...           ...           ...   
1018          0          1          Udyr       Warrior       Lee Sin   
1019          0          1           Jax       Warrior       Sejuani   
1020          0          1        Rumble  Mage-Warrior      Xin Zhao   
1021          0          1        Aatrox       Warrior      Xin Zhao   
1022          1          0   Mordekaiser       Warrior            Vi   

            t1p2_class t1p3_champion t1p3_class t1p4_champion  \
0              Warrior         Neeko       Mage       Caitlyn   
1              Warrior          Azir       Mage       Kalista   
2              Warrior          Azir       Mage         Senna   
3              Warrior          Hwei       Mage       Kalista   
4                 Tank      Tristana   Marksman        Lucian   
...                ...           ...        ...           ...   
1018  Warrior-Assassin         Akali   Assassin       Kalista   
1019              Tank          Azir       Mage         Varus   
1020           Warrior         Akali   Assassin  Miss Fortune   
1021           Warrior         Corki   Marksman         Xayah   
1022           Warrior          Azir       Mage        Lucian   

            t1p4_class t1p5_champion    t1p5_class t2p1_champion  \
0             Marksman           Lux          Mage        Aatrox   
1             Marksman  Renata Glasc       Support       K'Sante   
2     Marksman-Support        Maokai          Tank        Aatrox   
3             Marksman      Nautilus  Tank-Support        Darius   
4             Marksman         Milio       Support          Udyr   
...                ...           ...           ...           ...   
1018          Marksman          Hwei          Mage          Gnar   
1019          Marksman  Renata Glasc       Support       K'Sante   
1020          Marksman         Neeko          Mage       K'Sante   
1021          Marksman         Rakan       Support       K'Sante   
1022          Marksman         Milio       Support        Irelia   

        t2p1_class t2p2_champion        t2p2_class t2p3_champion  \
0          Warrior       Lee Sin  Warrior-Assassin          Azir   
1     Tank-Warrior      Nocturne          Assassin           Lux   
2          Warrior      Xin Zhao           Warrior         Sylas   
3     Tank-Warrior      Xin Zhao           Warrior         Akali   
4          Warrior       Lee Sin  Warrior-Assassin          Azir   
...            ...           ...               ...           ...   
1018       Warrior          Rell      Tank-Support        Kai'Sa   
1019  Tank-Warrior        Maokai              Tank       LeBlanc   
1020  Tank-Warrior       Lee Sin  Warrior-Assassin          Azir   
1021  Tank-Warrior           Jax           Warrior       Orianna   
1022       Warrior         Viego           Warrior       Orianna   

         t2p3_class t2p4_champion        t2p4_class t2p5_champion  \
0              Mage       Kalista          Marksman  Renata Glasc   
1              Mage         Ziggs              Mage         Janna   
2      Mage-Warrior        Ezreal     Mage-Marksman         Yuumi   
3          Assassin        Lucian          Marksman         Milio   
4              Mage       Kalista          Marksman  Renata Glasc   
...             ...           ...               ...           ...   
1018       Marksman         Varus          Marksman  Renata Glasc   
1019  Mage-Assassin       Kalista          Marksman          Zyra   
1020           Mage         Senna  Marksman-Support      Nautilus   
1021           Mage       Kalista   

Which shows the output csv file has the correct number of rows and columns as the merged_data dataframe.